### ISSUE TITLE
`podman search` doesn't honor [registries.insecure]

### ISSUE THREAD

**Issue by robodendron** [projectatomic/libpod#914]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**Description**

When I try to `podman search --registry localhost:5000 some/image:tag`, I get the usual HTTP/HTTPS mismatch error. Unfortunately there's no way to override that with `podman search --tls-verify` or `[registries.insecure]`.

**Steps to reproduce the issue:**

```
docker run -d -p 5000:5000 --restart=always --name registry registry:2
podman search --registry localhost:5000 some/image:tag
```

**Describe the results you received:**

`ERRO[0000] error searching registry "localhost:5000": couldn't search registry "localhost:5000": pinging docker registry returned: Get https://localhost:5000/v2/: http: server gave HTTP response to HTTPS client`

**Describe the results you expected:**

Podman honors the `[registries.insecure]` setting of /etc/containers/registries.conf and/or has a `--tls-verify` flag.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:       0.5.4
Go Version:    go1.10.2
OS/Arch:       linux/amd64
```

**Output of `podman info`:**

```
host:
  MemFree: 751910912
  MemTotal: 16652435456
  SwapFree: 8374448128
  SwapTotal: 8398041088
  arch: amd64
  cpus: 4
  hostname: xyz.example.com
  kernel: 4.16.13-300.fc28.x86_64
  os: linux
  uptime: 12h 42m 49.51s (Approximately 0.50 days)
insecure registries:
  registries:
  - localhost:5000
registries:
  registries:
  - localhost:5000
  - registry.fedoraproject.org
  - docker.io
store:
  ContainerStore:
    number: 24
  GraphDriverName: overlay
  GraphOptions:
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 41
  RunRoot: /var/run/containers/storage
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by rhatdan** [comment]
@haircommander PTAL

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @mtrmac Here is a case for insecure registries.

**Comment by mtrmac** [comment]
> @TomSweeneyRedHat @mtrmac Here is a case for insecure registries.

This is a reasonable use case, but if you are referring to the earlier conversation in https://github.com/projectatomic/buildah/pull/759 , I don’t think this is an argument for mixing the “insecure” and “search” lists.

**Comment by haircommander** [comment]
closed because #932 successful merge
